### PR TITLE
feat(snowflake): native pyarrow support

### PR DESCRIPTION
**Pull request by cpcloud**
This PR implements native `to_pyarrow` and `to_pyarrow_batches` methods for the snowflake backend.

### CODE REVIEW

File: ibis/backends/snowflake/__init__.py [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 
 import contextlib
+import itertools
 import json
 import warnings
-from typing import Any, Iterable, Mapping
+from typing import TYPE_CHECKING, Any, Iterable, Mapping
 
 import sqlalchemy as sa
+from pandas.core.flags import weakref
```
Comment (gforsyth) [inline ibis/backends/snowflake/__init__.py:10]: auto-importer looks to be on the fritz

**Review by gforsyth — CHANGES_REQUESTED** [review]

File: ibis/backends/snowflake/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/snowflake/__init__.py]: Yep, just noticed! Thanks.

**Review by gforsyth — APPROVED** [review]
This is great -- assuming it passes the test suite this looks good to go.

There's a reference in the snowflake-connector to an option for output called `arrow_force` that I want to know more about and also never want to know anything about.

**Comment by cpcloud** [comment]
Couple of tests failing locally, need to investigate them.

**Comment by cpcloud** [comment]
Figured it out, tests all passing now:

```
…/ibis on  snowflake-native-arrow is 📦 v5.0.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env)
❯ pytest -m snowflake -n auto --dist loadgroup --randomly-dont-reset-seed -q
bringing up nodes...
.xxx.x.......xx..x.xx....x...xsx...x.x......x..x............................x................x... [ 10%]
.....xx..........xxxxx.xxxx......xxx..xxxxx..xsx..x.x..x.x..x.x...........x..xx................x. [ 21%]
.x....xxx...x.x...x..x............x....x.x.......x.x..xxxxxx.x...x......x..s..xx........x...x.sx. [ 32%]
.......xx..x.......x..xx.........x..xx....x...x...xx..x................x.....x.xxx......x......x. [ 43%]
..x.....xx.x.x.x.x....x....x.x............xxx...........x.xx..................................xx. [ 54%]
....................x.x....x......x..x.......x.....x...................x.............x........x.. [ 65%]
.x.....x.x......x....x............................................x.x............................ [ 76%]
............x..x.x.......x..........x.x.......x.....................x.......................x.x.. [ 87%]
...................................................................x....x.xx.....x..x....x.x..... [ 98%]
...x.s.....                                                                                       [100%]
728 passed, 5 skipped, 151 xfailed in 97.40s (0:01:37)
```

**Pull request merged**